roadmap.md

```diff
@@ -0,0 +1,187 @@
+- [`Task` Versioning](https://github.com/tektoncd/pipeline/issues/1839) - Also required
+  for [Catalog](#catalog) success
+- [Alternative Task Implementations](https://github.com/tektoncd/pipeline/issues/215)
+- [Local Execution](https://github.com/tektoncd/pipeline/issues/235)
+- Testing and debugging frameworks
+- Emiting events throughout `Pipeline` execution
+- [Config as code](https://github.com/tektoncd/pipeline/issues/859) - This has some cross
+  over with [a related Triggers issue](https://github.com/tektoncd/triggers/issues/189)
+- [Performant Tekton](https://github.com/tektoncd/pipeline/issues/540) - We should set
+  performance requirements and measure against them
+- [Adding support for other architectures](https://github.com/tektoncd/pipeline/issues/856)
+- [Rich type support for params](https://github.com/tektoncd/pipeline/issues/1393)
+
+#### Task Interfaces and PipelineResources
+
+In alpha Tekton Pipelines, [`PipelineResources`](https://github.com/tektoncd/pipeline/issues/1673)
+were the interface between `Tasks` in a `Pipeline`: they were used for sharing typed
+data and variables between `Tasks`. This is being revisited and the feature has been
+deconstructed into:
+
+* `workspaces` - A way for a `Task` to declare data it needs and provides without
+  needing to know about the underlying mechanism used
+* `Task results` (aka "output params") - A way for a `Task` to provide values as outputs
+  which can be provided to downstream `Tasks`
+
+After adding these features we will once again reconsider the design of `PipelineResources`
+and decide what additional features, if any, we need.
+
+We will also need to decide how to continue to release and maintain the code that
+is used to build the images used by the existing `PipelineResources`:
+https://github.com/tektoncd/pipeline/tree/master/cmd.
+
+### Triggers
+
+In 2019 we created a simple system for creating instances of Tekton resources, triggered
+by json payloads sent to HTTP endpoints (Tekton Triggers).
+
+In 2020 we would like to add missing features and then push for
+[Triggers to be Beta and later GA](#beta-and-ga):
+
+* [Improved support for many EventListeners](https://github.com/tektoncd/triggers/issues/370)
+* [Pluggable core interceptors](https://github.com/tektoncd/triggers/issues/271)
+* [Support for expressions in TriggerBindings](https://github.com/tektoncd/triggers/issues/271)
+* [Using TriggerTemplates outside the context of an event](https://github.com/tektoncd/triggers/issues/200)
+* [Routing to multiple interceptors](https://github.com/tektoncd/triggers/issues/205)
+* [Dynamic TriggerTemplate parameters](https://github.com/tektoncd/triggers/issues/87)
+* [Performant Triggers](https://github.com/tektoncd/triggers/issues/406)
+* Support for poll-based triggering (e.g. when a repo changes state)
+* Support for additional expression languages
+* Increased traceability (e.g. why did my interceptor reject the event?)
```

Comment:
I'd say there are mainly two buckets of use cases that we are working on:

-- one for operator user type use cases (aka the listener part) e.g. how many triggers do I add to my eventListener, how do I scale with many event listeners, how do I expose the EL healthchecks, logs, add/remove custom interceptors etc.

-- end user use cases  i.e. trigger definitions:  expression languages,  built in functions (base64 decoding/ trimming commit SHAs), inline bindings etc.

Reply:
Okay I've tried to capture this in the newest version but you should feel free to just re-write it!